### DISCUSSION TITLE
Checked build builds libraries with MT runtime, even though NV_USE_DEBUG_WINCRT is set to True in CMake preset

### DISCUSSION THREAD

**Discussion by thfProjects** [NVIDIA-Omniverse/PhysX#78] [Q&A] [answered]
Hi, I'm new to C++ so I apologize if I'm not understanding something. 

I just spent a good amount of time getting my application to compile with Physx libraries. I built them with the checked build configuration as was recommended in the documentation. 
My Application uses MTd runtime, so in the CMake preset file, I made sure that I set NV_USEDEBUG_WINCRT and NV_USE_STATIC_WINCRT to true, hoping that this would build MTd lib files, but when linking these libs to my application I was always getting an error like this:

`LNK2038 mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MTd_StaticDebug`

When inspecting the properties of the generated Physx projects in Visual studio I found that they are actually being built with MT instead. 

Only when I choose the debug build configuration does it output MTd lib files, and then with them my application compiles.

So I am confused as to what the CMake preset file even does, because it seems like it's the build configuration presets that affect which CRT is used, and not the CMake preset. And I ran generate_projects.bat every time after making changes to the CMake preset.

**Comment by ayoub-belarbi** [comment, marked as answer]
Hello @thfProjects,

So "MTd" specify the runtime library that is used when building in Debug mode. That's what the "d" stands for.
What you did to set the MTd runtime is actually correct, however, again, it's used only for Debug mode and you can see that clearly in https://github.com/NVIDIA-Omniverse/PhysX/blob/release/104.1/physx/source/compiler/cmake/windows/CMakeLists.txt#L48.

The MTd runtime library includes additional functionalities that can help with debugging (extra error checking and runtime assertions) but will significantly increase the size and performance, which make sense for debug and not for release. In your case, wouldn't be possible to build your application using MT in release and MTd in debug? 

Thanks

**Reply by thfProjects** [reply]
Hey, thanks for the reply. That's all clear now.
I am generally familiar with debug vs release builds, but not the details of it for c++ and visual studio. 
It's not like I need to use MTd, but I am using it because it is early in development.
